packages/teams-js/src/public/mail.ts

```diff
@@ -14,6 +14,10 @@ export namespace mail {
         throw 'Not Supported';
       }
 
+      if (!openMailItemParams.itemId || !openMailItemParams.itemId.trim()) {
```

Comment:
How do we decide where it's OK to just let the TypeScript type system enforce stuff, and when we need to do our own validation on top of it? E.g. should you check if `openMailItemParams.itemId` is a `string` too before calling `trim()` on it?

Reply:
OK. That was a bit of a leading question I guess, and I hoped to get the very answer you gave. A follow up question - why do we then check for `!openMailItemParams.itemId`? It's already covered by the type system as `itemId` is a required property:

```
  export interface ComposeReplyOrForwardParams<T extends ComposeMailType> extends ComposeMailBase<T> {
    itemid: string;
  }
```

The `...trim()` check is a different story, as that's not something you can push to type system. But the undefined/null check is already covered.

Reply:
I think we could leave out the !openMailItemParams.itemId check if it makes us feel bad. Fwiw it is enforcing that someone can't create an OpenMailItemParams object that looks like this:

`{itemId: undefined}`

I think it's unlikely someone would do that, but having this check in place will prevent it explicitly (and I have a test that checks that)